### ISSUE TITLE
gazebo11 StaticMapPlugin not loading map tiles properly

### ISSUE THREAD

**Issue by Jaeyoung-Lim** [osrf/gazebo#2747]
When using the `StaticMapPlugin`, the plugin is able to download the map tiles, but gazebo doesn't seem to disable it properly. 
Terminal output: 
```
[Msg] Waiting for master.
[Msg] Connected to gazebo master @ http://127.0.0.1:11345
[Msg] Publicized address: 192.168.8.100
[Msg] Downloading map tile: tile_37.386832_-122.065684.png
[Msg] Downloading map tile: tile_37.386832_-122.065255.png
[Msg] Downloading map tile: tile_37.386832_-122.064826.png
[Msg] Downloading map tile: tile_37.386491_-122.065684.png
[Msg] Downloading map tile: tile_37.386491_-122.065255.png
[Msg] Downloading map tile: tile_37.386491_-122.064826.png
[Msg] Downloading map tile: tile_37.38615_-122.065684.png
[Msg] Downloading map tile: tile_37.38615_-122.065255.png
[Msg] Downloading map tile: tile_37.38615_-122.064826.png
```

![image](https://user-images.githubusercontent.com/5248102/83699501-7d456000-a604-11ea-80c3-c7e9d3d80c64.png)

- Usage followed instructions by [tuorial](http://gazebosim.org/tutorials?tut=static_map_plugin&cat=build_world)
- Related: https://answers.gazebosim.org/question/25083/gazebo-static-map-plugin-doesnt-work/?answer=25240#post-id-25240
- Gazebo 11.0.0 running on Ubuntu focal 20.04

**Comment by Jaeyoung-Lim** [comment]
@chapulina Would you have any idea what is going on? How can we fix this?

**Comment by chapulina** [comment]
I don't have much time to try to replicate this. Some ideas that may help:

* The console messages suggest that the tiles were correctly downloaded, do they look correct on their own? Are they valid images?
* Just making sure, no other messages are printed to the console? I'd expect some red errors.

**Comment by BaltashovIlia** [comment]
Hi, @chapulina 

I have the same problem. The problem for me was that my Google maps API key stopped working.

At the same time, StaticMapPlugin did not give any errors or warnings. I think it's worth adding a key validation check, if possible, of course.

**Comment by Jaeyoung-Lim** [comment]
@BaltashovIlia Did updating the keys solve the issue for you?

**Comment by BaltashovIlia** [comment]
Yes

**Comment by chapulina** [comment]
> I think it's worth adding a key validation check

Sounds like a good idea, [here](https://github.com/osrf/gazebo/blob/gazebo11/plugins/StaticMapPlugin.cc)'s the plugin code.

**Comment by Jaeyoung-Lim** [comment]
@BaltashovIlia I don't think we have the same issue then. Otherwise, I wouldn't have been able to download the map tiles

**Comment by BaltashovIlia** [comment]
@Jaeyoung-Lim 

Actually, if you have invalid Static Map API key then StaticMapPlugin will download tiles and even create png files. The problem is that you revived text with error message from google maps not image. Not the most obvious behavior )=.

To check it please open .gazebo/models/map_satellite_*/textures/*.img. If you could not open the file, then there is a high probability it was in the api key.

To check the api key, put this https request with your api key in the browser:

https://maps.googleapis.com/maps/api/staticmap?center=Albany,+NY&zoom=13&scale=1&size=600x300&maptype=roadmap&key=API_KEY&format=png&visual_refresh=true

**Comment by kyuhyong1** [comment]
I applied API restrictions to allow Maps Static API and it is working.
Restrict key
1 API
Selected APIs:
Maps Static API
![image](https://github.com/gazebosim/gazebo-classic/assets/145411927/d33d02e2-2e4b-4e69-ac37-268db8a83d9c)
